### PR TITLE
[7782] django upgrade

### PR DESCRIPTION
**Pull request by m4ra**
Some issues:

[django 4.0](https://docs.djangoproject.com/en/4.0/releases/4.0/#features-removed-in-4-0) set_language doesn't set the user language in request.session (key _language)
[django-cps](https://github.com/mozilla/django-csp) package has not been updated since 2yrs ago and gives deprecation warning:

> csp/utils.py:105: DeprecationWarning: child-src is deprecated in CSP v3. Use frame-src and worker-src

[django-multiselectfield](https://github.com/goinnn/django-multiselectfield) also appears unsupported/dead and gives error with django > 4.0. Fixed by adding max_length in respected models that use MultiSelectField

### CODE REVIEW

**Comment by goapunk** [comment]
sounds like we maybe should look into getting rid of django-multiselectfield :/

**Comment by m4ra** [comment]
> sounds like we maybe should look into getting rid of django-multiselectfield :/

@goapunk it's fixed by adding max_length.
Bcz if we get rid of this one, we shall also get rid of django-cps.. also not changed since 2yrs ago

File: apps/images/migrations/0006_alter_customrendition_file.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        ('apps_images', '0005_auto_20231025_1401'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='customrendition',
+            name='file',
+            field=wagtail.images.models.WagtailImageField(height_field='height', storage=wagtail.images.models.get_rendition_storage, upload_to=wagtail.images.models.get_rendition_upload_to, width_field='width'),
+        ),
+    ]
```
Comment (m4ra) [inline apps/images/migrations/0006_alter_customrendition_file.py:19]: @goapunk was that a missed one from wagtail upgrade?

Comment (goapunk) [reply apps/images/migrations/0006_alter_customrendition_file.py:19]: no, it isn't created on main. So must be something related to django

File: changelog/7782.md [hunk 2]
```diff
@@ -8,3 +8,8 @@
     - Django from 4.0 to 4.1
       - apps:  added max_length in multiselectfield to fix index error with
         validators[0]=MaxValueMultiFieldValidator(self.max_length)
+      - apps:  added migration for wagtail custom renditions
+
+    - Django from 4.1 to 4.2
+      - site-packages: update taggit to v4. 
+        Should update to latest version when compatible with wagtail to fix index_together deprecation
```
Comment (goapunk) [inline changelog/7782.md:15]: is this only compatible with wagtail 5.2 or why do we need to wait ?

**Review by goapunk — COMMENTED** [review]
Looks good to me, just some questions

File: changelog/7782.md (hunk 2 above)
Comment (m4ra) [reply changelog/7782.md:15]: yes, current wagtail 5.1.3 compatible to taggit > 5.0

**Review by philli-m — APPROVED** [review]
Nice! quick click around and multi select field adding is all working smooth, couple questions but I think it's good to go, as we wait anyway for @goapunk PR will leave merge to him

File: digitalstrategie/templates/includes/upper_footer.html [hunk 3]
```diff
@@ -22,7 +22,8 @@ <h3 class="heading js-trigger">{{ menuitem.title }}</h3>
                                 {% get_language_info for 'de-ls' as language %}
                                 {% get_has_translation page 'de_ls' as has_translation_de_ls %}
                                 <li>
-                                    <form class="footer-menu__btn" action="{% url 'set_language' %}" method="post">
+					<!-- following might not working bcz of django 4.0 doesn’t set the user language in request.session--> 
+                                    <form class="footer-menu__btn" action="{% url 'set_language' %}" method="post"
```
[the remark this answers is not in the document]
Comment (m4ra) [reply digitalstrategie/templates/includes/upper_footer.html]: hmm don't remember why i put following there, but from release 4.0:
> django.views.i18n.set_language() doesn’t set the user language in request.session (key _language).
ref: https://docs.djangoproject.com/en/4.0/releases/4.0/#features-removed-in-4-0
this url only gets added when language is german, but I don't get what's doing, as it is also hidden in the footer.

[the remark this answers is not in the document]
Comment (m4ra) [reply digitalstrategie/templates/includes/upper_footer.html]: I removed that comment, but can put it back in a more readable language :)

File: changelog/7782.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  - templates: ifequal not supported 
+
+- Django from 4.0 to 4.1 (#7782)
+  - apps:  added max_length in multiselectfield to fix index error with
+    validators[0]=MaxValueMultiFieldValidator(self.max_length)
+  - apps:  added migration for wagtail custom renditions
+
+- Django from 4.1 to 4.2 (#7782)
+  - site-packages: update taggit to v4 
+    Should update to latest version when compatible with wagtail to fix index_together deprecation
```
Comment (m4ra) [inline changelog/7782.md:15]: @goapunk unfortunaltely 5.2 still only compatible with taggit < 5.0

**Comment by m4ra** [comment]
> some notes:
allowed_hosts needs update in admin once merged, see https://docs.djangoproject.com/en/4.2/releases/4.0/#csrf-trusted-origins-changes

@goapunk so we cannot merge yet bcz admin repo stalk update messed it up?

**Comment by goapunk** [comment]
> > some notes:
> > allowed_hosts needs update in admin once merged, see https://docs.djangoproject.com/en/4.2/releases/4.0/#csrf-trusted-origins-changes
> 
> @goapunk so we cannot merge yet bcz admin repo salt update messed it up?

no, should be ok to merge, these kind of changes still work

File: requirements/prod.txt [hunk 5]
```diff
@@ -1,2 +1,2 @@
 -r base.txt
-psycopg2==2.9.9
+psycopg==3.1.12
```
[the remark this answers is not in the document]
Comment (m4ra) [reply requirements/prod.txt:2]: [c]?

File: requirements/base.txt [hunk 6]
```diff
@@ -1,11 +1,13 @@
 brotli==1.1.0
-Django==3.2.19
+Django==4.2
 django-cloudflare-push==0.2.2
 django-csp==3.7
+django-filter==23.3
```
[the remark this answers is not in the document]
Comment (m4ra) [reply requirements/base.txt:5]: wagtail

File: requirements/base.txt [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 django-multiselectfield==0.1.12
+django-taggit==4.0.0
```
[the remark this answers is not in the document]
Comment (m4ra) [reply requirements/base.txt:7]: Ic, where should I add the comment then?

File: requirements/prod.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (goapunk) [reply requirements/prod.txt:2]: see https://www.psycopg.org/psycopg3/docs/basic/install.html#local-installation

File: requirements/base.txt (hunk 7 above)
[the remark this answers is not in the document]
Comment (goapunk) [reply requirements/base.txt:7]: above the line ? :)

[the remark this answers is not in the document]
Comment (m4ra) [reply requirements/base.txt:7]: it comes with wagtail, so I will remove it.

**Review by goapunk — APPROVED** [review]
LGTM

**Pull request merged**